src/Transports.AspNetCore/WebSockets/GraphQLWs/MessageType.cs

```diff
@@ -0,0 +1,155 @@
+using GraphQL.Transport;
+
+namespace GraphQL.Server.Transports.AspNetCore.WebSockets.GraphQLWs;
+
+/// <summary>
+/// Protocol message types defined in
+/// <see href="https://github.com/enisdenjo/graphql-ws/blob/master/PROTOCOL.md"/>
+/// and
+/// <see href="https://github.com/enisdenjo/graphql-ws/blob/master/src/common.ts"/>
+/// </summary>
+public static class MessageType
+{
+    /// <summary>
+    /// <para>
+    /// Indicates that the client wants to establish a connection within the existing socket.
+    /// This connection is not the actual WebSocket communication channel, but is rather a
+    /// frame within it asking the server to allow future operation requests.
+    /// </para>
+    /// <para>
+    /// The server must receive the connection initialization message within the allowed waiting
+    /// time specified in the <see cref="GraphQLWebSocketOptions.ConnectionInitWaitTimeout"/>
+    /// parameter during the server setup.  If the client does not request a connection within
+    /// the allowed timeout, the server will close the socket with the event:
+    /// <i>4408: Connection initialization timeout</i>.
+    /// </para>
+    /// <para>
+    /// If the server receives more than one <see cref="ConnectionInit"/> message at any
+    /// given time, the server will close the socket with the event
+    /// <i>4429: Too many initialization requests</i>.
+    /// </para>
+    /// <para>
+    /// The client may use the optional <see cref="OperationMessage.Payload"/> field to transfer
+    /// additional details about the connection.
+    /// </para>
+    /// </summary>
+    public const string ConnectionInit = "connection_init"; // Client -> Server
+
+    /// <summary>
+    /// <para>
+    /// Expected response to the <see cref="ConnectionInit"/> message from the client acknowledging
+    /// a successful connection with the server.
+    /// </para>
+    /// <para>
+    /// The server can use the optional <see cref="OperationMessage.Payload"/> field to transfer additional details about the connection.
+    /// </para>
+    /// <para>
+    /// The client is now ready to request subscription operations.
```

Comment:
Just a note - not only subscriptions operations.

Reply:
Correct.